### PR TITLE
Always cleanup, load a library with @date and fix dashboards.

### PR DESCRIPTION
**Pull request by dblock**
Signed-off-by: dblock <dblock@amazon.com>

### Description

Another build failed because it was missing $TOKEN, which was changed previously in https://github.com/opensearch-project/opensearch-build/pull/1101. The logs also contain that commit. The only explanation is that the change was not picked up from `vars`, meaning the library was loaded and not replaced. 

Attempting to clean the workspace every time to fix that and to use a @date-based reference to force re-loading.

Forgot to use new build lib code in dashboards builds everywhere. 

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by gaiksaya** [comment]
Even if we do that I see this log `[WS-CLEANUP] Deferred wipeout is disabled by the job configuration...` looks like we need to enable it from job config?

**Comment by dblock** [comment]
> Deferred wipeout

Apparently.

File: jenkins/opensearch-dashboards/Jenkinsfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-lib = library(identifier: "jenkins@current", retriever: legacySCM(scm))
+lib = library(identifier: "jenkins@20211118", retriever: legacySCM(scm))
```
Comment (gaiksaya) [inline jenkins/opensearch-dashboards/Jenkinsfile:1]: Where is this library retrieved from?

Comment (dblock) [reply jenkins/opensearch-dashboards/Jenkinsfile:1]: legacySCM means basically local checked out workspace, but it also caches things in-memory in the service instance and i don't know what the reloading rules are. I suspect it asks "has anything changed" to the "localSCM", which is git, or at least it should.

**Review by gaiksaya — APPROVED** [review]

**Review by owaiskazi19 — APPROVED** [review]

**Pull request merged**